xarray/testing.py

```diff
@@ -42,6 +42,8 @@ def assert_equal(a, b):
         The first object to compare.
     b : xarray.Dataset, xarray.DataArray or xarray.Variable
         The second object to compare.
+    name : str, optional
```

Comment:
Maybe we should have a private function, e.g., `assert_variable_equal`, that accepts an optional `name` and call that from `assert_equal`? It's a little weird to allow names for Dataset objects.

Reply:
I didn't consider it as a name for either object, but rather an identifier for the specific assertion. Maybe it should be named/documented better as such?